de.dlr.sc.overtarget.language/src/de/dlr/sc/overtarget/language/generator/OvertargetGenerator.xtend

```diff
@@ -28,10 +31,21 @@ import de.dlr.sc.overtarget.language.targetmodel.TargetmodelPackage
  */
 class OvertargetGenerator extends AbstractGenerator {
 	static val DEFAULT_JRE_CONTAINER = "org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/";
-	
+
 	override void doGenerate(Resource resource, IFileSystemAccess2 fsa, IGeneratorContext context) {
 		for (model : resource.allContents.toIterable.filter(TargetModel)) {
-			fsa.generateFile(model.name + ".target", OvertargetOutputConfigurationProvider.GENERATOR_OUTPUT_ID_OVERTARGET, model.compile)
+			EcoreUtil.resolveAll(resource)
+			if (ReferenceTargetHelper.importedModelIsProxy(model) == true || ReferenceTargetHelper.parentIsProxy(model) == true) {
+				ReferenceTargetHelper.getModelToGenerate(model)
```

Comment:
It's confisuing that the method name is get* but it doesn't return the model... Maybe leave the model object itself as it is and return the updated model?   -> Getters usally always return the value you are trying to get